models/DataObject/ClassDefinition/Data/Block.php

```diff
@@ -26,7 +26,7 @@
 use Pimcore\Normalizer\NormalizerInterface;
 use Pimcore\Tool\Serialize;
 
-class Block extends Data implements CustomResourcePersistingInterface, ResourcePersistenceAwareInterface, LazyLoadingSupportInterface, TypeDeclarationSupportInterface, VarExporterInterface, NormalizerInterface, DataContainerAwareInterface, PreGetDataInterface, PreSetDataInterface
+class Block extends Data implements CustomResourcePersistingInterface, ResourcePersistenceAwareInterface, LazyLoadingSupportInterface, TypeDeclarationSupportInterface, VarExporterInterface, NormalizerInterface, DataContainerAwareInterface, PreGetDataInterface, PreSetDataInterface, ClassSavedInterface
```

Comment:
@Corepex Make it sense to add `DataContainerAwareInterface` and `ClassSavedInterface` in one class? I think we should only set once. So we can remove the `!DataContainerAwareInterface` check in 11.x

Reply:
Do you mean to merge the `DataContainerAwareInterface` and the `ClassSavedInterface` into one file or to remove the `DataContainerAwareInterface` completely from that class?

`DataContainerAwareInterface` implements `preSave` and `postSave`
`ClassSavedInterface` implements `classSaved`

Reply:
But the classSaved method of `ClassSavedInterface` will only executed if the class is not a instance of `DataContainerAwareInterface`. So I would not add it to this classes because it have no effect. We can remove the `!DataContainerAwareInterface` in 11.x when we only add it to non `DataContainerAwareInterface` classes.

Reply:
But for example, here we call the `preSave` method on each $fielddefinition and therefore, we check for the `DataContainerAwareInterface` to verify the method - this $fielddefinition could also be a `block` or?

https://github.com/pimcore/pimcore/blob/c8e3a921b3dca7779a6ae951c65185a922b763ed/models/DataObject/ClassDefinition.php#L426-L435

Reply:
You should not remove the `DataContainerAwareInterface` but the `ClassSavedInterface` interface

Reply:
Ah, sorry now I get it - that makes sense true 👍

Reply:
> !DataContainerAwareInterface

updated the follow-up ticket to also remove the `!DataContainerAwareInterface` check